wagtail/core/models.py

```diff
@@ -1189,10 +1202,52 @@ def specific(self):
             # reverted before switching branches). So, the best we can do is
             # return the page in it's current form.
             return self
+
         if isinstance(self, model_class):
             # self is already the an instance of the most specific class
             return self
-        return self.cached_content_type.get_object_for_this_type(id=self.id)
+
+        if deferred:
+            # Generate a tuple of values in the order expected by __init__(),
+            # with missing values substituted with DEFERRED ()
+            values = tuple(
+                getattr(self, f.attname, self.pk if f.primary_key else DEFERRED)
+                for f in model_class._meta.concrete_fields
+            )
+            # Create object from known attribute values
+            specific_obj = model_class(*values)
+            specific_obj._state.adding = self._state.adding
+        else:
+            # Fetch object from database
+            specific_obj = model_class._default_manager.get(id=self.id)
+
+        # Copy additional attribute values
+        if copy_attrs:
+            for attr in copy_attrs:
+                if attr in self.__dict__:
+                    setattr(specific_obj, attr, getattr(self, attr))
+
+        return specific_obj
+
+    @cached_property
+    def specific(self):
+        """
+        A wrapper for ``self.get_specific()`` that returns this page in its
```

Comment:
Could we remove "A wrapper for ``self.get_specific()`` that" from these two methods? Feel like jumping straight into what it does makes it a bit easier to read.